c2rust-analyze/src/rewrite/expr/convert.rs

```diff
@@ -477,55 +301,34 @@ fn materialize_adjustments<'tcx>(
     }
 }
 
-pub fn gen_hir_rewrites<'tcx>(
+/// Convert the MIR rewrites attached to each HIR node into `Span`-based `rewrite::Rewrite`s.
+pub fn convert_rewrites<'tcx>(
     tcx: TyCtxt<'tcx>,
-    mir: &Body<'tcx>,
     hir_body_id: hir::BodyId,
-    rewrites: &HashMap<Location, Vec<MirRewrite>>,
+    mir_rewrites: HashMap<HirId, Vec<mir_op::RewriteKind>>,
 ) -> Vec<(Span, Rewrite)> {
-    // Build `span_index`, which maps `Span`s to MIR `Locations`.
-    let span_index = build_span_index(mir);
-
     // Run the visitor.
     let typeck_results = tcx.typeck_body(hir_body_id);
     let hir = tcx.hir().body(hir_body_id);
 
-    let mut v = HirRewriteVisitor {
+    let mut v = ConvertVisitor {
         tcx,
         typeck_results,
-        mir,
-        span_index,
-        rewrites,
-        locations_visited: HashSet::new(),
-        hir_rewrites: Vec::new(),
+        mir_rewrites,
+        rewrites: Vec::new(),
         materialize_adjustments: false,
     };
     intravisit::Visitor::visit_body(&mut v, hir);
```

Comment:
Why not just

```suggestion
    v.visit_body(hir);
```

?  Is there a reason?

Reply:
This was copied from the previous version.  I changed the code (in both `unlower` and `convert`) to import `intravisit::Visitor` and call `v.visit_body(hir)` and there weren't any problems.